docstore/doc.go

```diff
@@ -186,23 +186,27 @@
 // Revisions
 //
 // Docstore supports document revisions to distinguish different versions of a
-// document and enable optimistic locking. By default, Docstore stores the revision
-// in the field named "DocstoreRevision" (stored in the constant
+// document and enable optimistic locking. By default, Docstore stores the
+// revision in the field named "DocstoreRevision" (stored in the constant
 // DefaultRevisionField). Providers give you the option of changing that field
 // name.
 //
-// Docstore gives every document a revision at creation time and changes that
-// revision whenever the document is modified. If modification methods are called
-// on a document struct or map without a revision field, then no revision
-// checks are performed, and changes are forced blindly. Otherwise, whenever
-// you pass a document with a revision to Put, Replace, Update or Delete,
-// docstore will compare the revision of the stored document to that of the
-// given document and return an error with code FailedPrecondition on mismatch.
-// (See https://gocloud.dev/gcerrors for information about error codes.)
-// If you call Get to retrieve a document with a revision, then later perform a
-// write action with that same document, it will fail if the document was changed
-// since the Get. Create, Put, Replace and Update all populate the revision
-// field of the argument document with the new revision value.
+// Docstore gives every document a revision at creation time and updates that
+// revision whenever the document is modified if and only if the document struct
+// or map has a revision field/key matching the revision field name. If you
```

Comment:
Remove the word "key" here. We  call them "fields" even if they're map keys. The word "key" is used for the document's unique ID.